### ISSUE TITLE
USB Printer Encoding Problem with Chinese Characters

### ISSUE THREAD

**Issue by DebianC** [python-escpos/python-escpos#281]
thank you for seeing my problem. 
There is something wrong with my encoding, it occurs that Chinese characters are wrongly printed when I write like this :
`printer.text("你好")`
but I get wrong characters printed like "挂批"
I tried to set the encoding as #-- coding: gbk -- ,but got encoding problem.
And I tried to use the magic encode like `from escpos.magicencode import Encoder`, but got `no module named escpos.magicencode` as ImportError.
So I wander how to set the encoding to make my printer print correct Chinese characters.
Sorry for my English and naive questions. 
I have been working on this for one week and still could't figure it out.
Thanks again.

**Comment by mike42** [comment]
Would you mind sharing which make/model of printer you have? This is important, because ESC/POS does not support Chinese text, but some printers have their own commands which you might be able to use.

**Comment by DebianC** [comment]
Thank you for your information! My printer model is Rongta RP325 bought in China. Could you tell me whether this printer can work or show me the list of printers which support Chinese text? If my printer cannot work, I'd appreciate it if you recommend me of one useful printer. And I wonder how can I learn to use the commands that the right printers have? So many questions…Thank you very much!

**Comment by slyang-git** [comment]
@DebianC try set the `codepage` parameter, worked for me to print Chinese characters:

```
from escpos.printer import Usb

usb_printer = Usb(0x8866, 0x0100, 0, 0x81, 0x02)
usb_printer.codepage = 'gb2312'

```

**Comment by suule95** [comment]
Can I set multiple codepages?

**Comment by slyang-git** [comment]
> Can I set multiple codepages?
> Especially for cases where there might be a mixture of traditional and simplified chinese.

maybe you can change codepage by if...else logics on the runtime.

**Comment by saper** [comment]
> Can I set multiple codepages?

Yes, you can, if the printer profile is known - see example https://github.com/receipt-print-hq/escpos-printer-db/blob/fe07c81155a6c49912379fdb960dc9155c8dfc89/data/profile/TM-T88IV.yml#L23-L34 - this module will know how to switch code pages automatically.
